internal/template/function.go

```diff
@@ -82,3 +84,50 @@ func resourceRequests(pods corev1.PodList, weights string) (float64, error) {
 	}
 	return totalResources, nil
 }
+
+func cpuUtilization(labelArgs ...string) (string, error) {
+
+	cpuUtilizationQueryTemplate := `
+scalar(
+  sum(
+    max(container_cpu_usage_seconds_total{container="", image=""}) by (pod)
+    *
+    on (pod) group_left kube_pod_labels{{ . }}
+  )
+  /
+  sum(
+    sum_over_time(kube_pod_container_resource_limits_cpu_cores[1h:1s])
+    *
+    on (pod) group_left kube_pod_labels{{ . }}
+  )
+)`
+
+	tmpl, err := template.New("query").Parse(cpuUtilizationQueryTemplate)
+	if err != nil {
+		return "", err
+	}
```

Comment:
`tmpl := template.Must(template.New("query").Parse(cpuUtilizationQueryTemplate))`

Reply:
A panic here felt too heavy.
.. granted, it's an internal packate/template so this shouldn't ever fail during normal run conditions.

Reply:
This is exactly when you _should_ panic: if the code itself is invalid (the constant doesn't parse) and the only action is to fix the code.

Reply:
Sorry, meant more in line with when this gets called it feels too late in the lifecycle.

I think more appropriately we should call the `template.Must` in an init function so we could panic early and reuse the template instead of creating a new one each time. But that felt like a larger change/discussion since we have other instances of template.New being called inline and I didn't want to distract from this. 

The above are more concerns/refactors for later, I'm fine with changing to template.must inline here.